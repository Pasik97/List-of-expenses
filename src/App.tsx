import React from 'react';
import styled from 'styled-components';
import Header from './modules/Header/Header';
import AddTransactionForm from './modules/AddTransactionForm/AddTransactionForm';
import StoreProvider from './modules/StoreProvider/StoreProvider';
import TransactionsList from './modules/TransactionsList/TransactionsList';
import TransactionsSummary from './modules/TransactionsSummary/TransactionsSummary';
import Stores from './stores/Stores';
import media from './utils/media';

const AppWrapper = styled.div`
   margin: 0 8px;

   ${media.fromTablet} {
      margin: 0 128px;
   }

   ${media.fromDesktop} {
      margin: 0 25%;
   }
`;

const store = new Stores();

export const App: React.FC = () => (
   <StoreProvider store={store}>
      <AppWrapper>
         <Header />
         <AddTransactionForm />
         <TransactionsList />
         <TransactionsSummary />
      </AppWrapper>
   </StoreProvider>
)

export default App;
