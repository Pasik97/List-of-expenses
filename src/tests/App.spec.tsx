import { create } from 'react-test-renderer';
import App from '../App';

jest.mock('modules/Header/Header');
jest.mock('modules/AddTransactionForm/AddTransactionForm');
jest.mock('modules/TransactionsList/TransactionsList');
jest.mock('modules/TransactionsSummary/TransactionsSummary');

describe('<App />', () => {
   it('should render Header, AddTransactionForm, TransactionsList, TransactionsSummary components', () => {
      const component = create(<App />);

      expect(component.toJSON()).toMatchSnapshot();
   });
});
