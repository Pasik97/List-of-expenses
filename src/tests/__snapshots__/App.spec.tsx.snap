// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreProvider /> should render Header, AddTransactionForm, TransactionsList, TransactionsSummary components 1`] = `
<div
  className="sc-bdvvtL fZgyCL"
>
  (Header /)
  (AddTransactionForm /)
  (TransactionsList /)
  (TransactionsSummary /)
</div>
`;
